mesos-slave: Docker optional & IP address configurable

ragnard opened the pull request:
Two minor improvements:
- Usage of docker containerizer is currently hardcoded, this PR makes it optional.
- Make IP address to listen on part of service configuration

wmertens commented on nixos/modules/services/misc/mesos-slave.nix:
```diff
@@ -22,8 +24,14 @@ in {
+
       port = mkOption {
-        description = "Mesos Slave port";
+        description = "Port to listen on.";
```
I prefer the previous description...

ragnard replied:
Thought mentioning "Mesos Slave" in descriptions in the mesos-slave service was redundant but not something I feel strongly about. Can revert.

wmertens commented on nixos/modules/services/misc/mesos-slave.nix:
```diff
@@ -43,6 +51,12 @@ in {
         type = types.bool;
       };
 
+      withDocker = mkOption {
```
How about using the docker service enable flag?

ragnard replied:
@wmertens That assumes you want to run the docker containerizer as soon as you also have the docker service enabled, which is not necessarily true (not in my case anyway). Is there a more elegant nix way to do this so that:

- default is to enable docker containerizer if docker service is enabled
- that behaviour is overridable, ie. you can force disabling docker containerizer

wmertens replied:
I think you can `mkDefault cfg.services.docker.enable`.

On Sun, Apr 12, 2015, 12:00 AM Ragnar Dahlén <notifications@github.com>
wrote:

> In nixos/modules/services/misc/mesos-slave.nix
> <https://github.com/NixOS/nixpkgs/pull/7302#discussion_r28198819>:
>
> > @@ -43,6 +51,12 @@ in {
> >          type = types.bool;
> >        };
> >
> > +      withDocker = mkOption {
>
> @wmertens <https://github.com/wmertens> That assumes you want to run the
> docker containerizer as soon as you also have the docker service enabled,
> which is not necessarily true (not in my case anyway). Is there a more
> elegant nix way to do this so that:
>
>    - default is to enable docker containerizer if docker service is
>    enabled
>    - that behaviour is overridable, ie. you can force disabling docker
>    containerizer
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/7302/files#r28198819>.
>

rushmorem replied:
@ragnard yes you can achieve both of those requirements using `config.virtualisation.docker.enable`. What's the status of this pull request? Still working on this? This is something I'm interested in as well.

ragnard replied:
@rushmorem I'm still interested too, just haven't had time. Will pick it up, sorry for the delay.

rushmorem replied:
@ragnard No problem. Let me know if you need any extra hands on this.

ragnard replied:
@rushmorem @wmertens  I've updated the PR. Didn't use `mkDefault` as I couldn't figure out how to use that properly in this context. However, just setting `default = config.virtualisation.docker.enable` seems to work. With that I think the state of this PR is good.

rushmorem replied:
@ragnard Thanks! This looks good to me as well. @wmertens @offlinehacker May we please have this merged?

ragnard committed:
mesos-slave: docker and IP address config options

- Usage of docker containerizer is currently hardcoded, this PR makes it
  optional. Default is to enable it if docker is enabled.
- Make IP address to listen on part of service configuration.

wmertens commented:
[![LGTM](http://lgtm.in/p/Ikvr4yRWp)](http://lgtm.in/i/Ikvr4yRWp)

wmertens commented:
@lethalman should the fact that docker is no longer required for mesos be mentioned in the release notes?
